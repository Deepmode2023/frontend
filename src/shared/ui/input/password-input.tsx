import { useState } from "react";
import InputAdornment from "@mui/material/InputAdornment";
import { BaseInput, type IBaseInput } from "./base-input";
import VisibilityIcon from "@mui/icons-material/Visibility";
import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";

const PasswordInput = (props: IBaseInput<"password">) => {
  const [isVisible, setIsVisible] = useState(false);
  return (
    <BaseInput
      type={isVisible ? "text" : "password"}
      slotProps={{
        input: {
          endAdornment: (
            <InputAdornment
              position="start"
              onClick={() => {
                setIsVisible((prev) => !prev);
              }}
            >
              {isVisible ? <VisibilityIcon /> : <VisibilityOffIcon />}
            </InputAdornment>
          ),
        },
      }}
      {...props}
    />
  );
};

export { PasswordInput };
