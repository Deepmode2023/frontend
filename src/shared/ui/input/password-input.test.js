import { describe, test, expect, beforeEach, jest } from "bun:test";
import { PasswordInput } from "./password-input";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import * as matchers from "@testing-library/jest-dom/matchers";

expect.extend(matchers);

describe("Testing 'PasswordInput' component", () => {
  let component;
  const registerOptions = {
    name: "testName",
    onChange: jest.fn(),
    onBlur: jest.fn(),
    ref: jest.fn(),
  };
  beforeEach(() => {
    component = render(
      <PasswordInput
        {...registerOptions}
        label="Test Label"
        formError={undefined}
      />
    );
  });

  test("... & icon click event", async () => {
    const input = component.container.querySelector("input");
    const iconButton = screen.getByTestId("password_input_icon");

    expect(screen.getByLabelText("Test Label")).toBeInTheDocument();
    expect(input).toHaveAttribute("type", "password");
    await userEvent.click(iconButton);
    expect(input).toHaveAttribute("type", "text");
    await userEvent.click(iconButton);
    expect(input).toHaveAttribute("type", "password");
  });

  test("... & make snapshot", () => {
    expect(component.container.innerHTML).toMatchSnapshot();
  });
});
