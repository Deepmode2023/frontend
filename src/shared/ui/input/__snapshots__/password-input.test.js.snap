// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing 'PasswordInput' component ... & make snapshot 1`] = `"<div data-testid="base_input" class="flex flex-col gap-1"><div class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"><label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard css-19fz36k-MuiFormLabel-root-MuiInputLabel-root" data-shrink="false" for="input-with-sx" id="input-with-sx-label">Test Label</label><div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-r5in8e-MuiInputBase-root-MuiInput-root"><input aria-invalid="false" id="input-with-sx" name="testName" type="password" class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1o3an4k-MuiInputBase-input-MuiInput-input" value=""><div class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-6ga8lj-MuiInputAdornment-root" data-testid="password_input_icon"><span class="notranslate">â€‹</span><svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root" focusable="false" aria-hidden="true" viewBox="0 0 24 24" data-testid="VisibilityOffIcon"><path d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7M2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2m4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3z"></path></svg></div></div></div></div>"`;
